### PR TITLE
ON-41012 # prevent auto save of disabled form inside throttledAutoSave

### PR DESCRIPTION
**Pull request by divporter**

### CODE REVIEW

File: src/hooks/useFormSubmissionAutoSaveState.ts [hunk 1]
```diff
@@ -64,27 +64,36 @@ export default function useFormSubmissionAutoSaveState({
...
-              return autoSaveService.upsertAutoSaveData<FormElement>(
-                definition.id,
-                `LAST_ELEMENT_UPDATED_${autoSaveKey}`,
-                lastElementUpdated,
-              )
+        if (!formIsDisabled) {
+          switch (lastElementUpdated?.type) {
+            case 'summary':
+            case 'calculation':
+            case 'captcha': {
```
Comment (mymattcarroll) [inline src/hooks/useFormSubmissionAutoSaveState.ts:71]: This wont prevent the `resetRecaptchas()` function from running as it does not set the `lastElementUpdated` property when updating the submission.

Comment (divporter) [reply src/hooks/useFormSubmissionAutoSaveState.ts:71]: yeah but this function will run nearly 10 seconds later, by which time it will have lost the race.

Comment (divporter) [reply src/hooks/useFormSubmissionAutoSaveState.ts:71]: hmmmm but memoing....

Comment (divporter) [reply src/hooks/useFormSubmissionAutoSaveState.ts:71]: Just tested with a 5 second delay on disabling the form and it worked.
With a 15 second delay it obviously triggered autosave. So I'm happy with this.

Comment (mymattcarroll) [reply src/hooks/useFormSubmissionAutoSaveState.ts:71]: As discussed, im happy if you are happy.

**Pull request merged**